File: server/app/Models/Movie.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Movie extends Model
{
    use HasFactory;

    protected $fillable = [
        'title', 'description', 'release_date', 'poster_image', 'backdrop_image',
        'type', 'duration', 'seasons', 'episodes', 'director', 'cast',
        'tmdb_id', 'average_rating', 'vote_count'
    ];

    protected $casts = [
        'cast' => 'array',
        'release_date' => 'date',
    ];

    public function genres()
    {
        return $this->belongsToMany(Genre::class, 'movie_genre');
    }

    public function ratings()
    {
        return $this->hasMany(Rating::class);
    }

    public function reviews()
    {
        return $this->hasMany(Review::class);
    }

    public function watchlists()
    {
        return $this->hasMany(Watchlist::class);
    }

    // Method to update average rating after new rating
    public function updateAverageRating()
    {
        $this->average_rating = $this->ratings()->average('rating') ?? 0;
        $this->vote_count = $this->ratings()->count();
        $this->save();
    }
}

File: movie_app/lib/app/data/models/movie.dart
class MovieItem {
  final int id;
  final String title;
  final String? description;
  final String? releaseDate;
  final String? posterImage; // <- nullable
  final String? backdropImage;
  final String type;
  final int duration;
  final int seasons;
  final int episodes;
  final String? director;
  final List<String>? cast;
  final int tmdbId;
  final double averageRating;
  final int voteCount;
  final String category;
  final String status;

  MovieItem({
    required this.id,
    required this.title,
    this.description,
    this.releaseDate,
    this.posterImage,
    this.backdropImage,
    required this.type,
    required this.duration,
    required this.seasons,
    required this.episodes,
    this.director,
    this.cast,
    required this.tmdbId,
    required this.averageRating,
    required this.voteCount,
    required this.category,
    required this.status,
  });

  factory MovieItem.fromJson(Map<String, dynamic> json) => MovieItem(
        id: json['id'],
        title: json['title'],
        description: json['description'],
        releaseDate: json['release_date'],
        posterImage: json['poster_image'],
        backdropImage: json['backdrop_image'],
        type: json['type'],
        duration: json['duration'],
        seasons: json['seasons'],
        episodes: json['episodes'],
        director: json['director'],
        cast: List<String>.from(json['cast'] ?? []),
        tmdbId: json['tmdb_id'],
        averageRating: (json['average_rating'] ?? 0).toDouble(),
        voteCount: json['vote_count'] ?? 0,
        category: json['category'],
        status: json['status'],
      );
}
